sklearn/model_selection/_validation.py

```diff
@@ -492,6 +493,43 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
     return ret
 
 
+def _score_single_metric(estimator, X_test, y_test, scorer):
```

Comment:
added a new function which runs scorer once. earlier the same code was being repeated in _score and _multimetricscore